The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6977816142/job/18988199797) [(plain)](https://github.com/rust-lang/rust/commit/c2b58d7d0b5fc434d861f29a458da3c31dcd2977/checks/18988199797/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
---- library/core/src/ptr/const_ptr.rs - ptr::const_ptr::*const[T]::is_empty (line 1651) stdout ----
error[E0433]: failed to resolve: use of undeclared crate or module `ptr`
##[error] --> library/core/src/ptr/const_ptr.rs:1654:26
  |
6 | let slice: *const [i8] = ptr::slice_from_raw_parts(ptr::null(), 3);
  |                          ^^^ use of undeclared crate or module `ptr`
help: a builtin type with a similar name exists
  |
  |
6 | let slice: *const [i8] = str::slice_from_raw_parts(ptr::null(), 3);
help: consider importing one of these items
  |
5 + use core::ptr;
  |
  |
5 + use std::ptr;
  |

error[E0433]: failed to resolve: use of undeclared crate or module `ptr`
##[error] --> library/core/src/ptr/const_ptr.rs:1654:52
  |
6 | let slice: *const [i8] = ptr::slice_from_raw_parts(ptr::null(), 3);
  |                                                    ^^^ use of undeclared crate or module `ptr`
help: a builtin type with a similar name exists
  |
  |
6 | let slice: *const [i8] = ptr::slice_from_raw_parts(str::null(), 3);
help: consider importing one of these items
  |
5 + use core::ptr;
  |
---
---- library/core/src/ptr/mut_ptr.rs - ptr::mut_ptr::*mut[T]::is_empty (line 1921) stdout ----
error[E0433]: failed to resolve: use of undeclared crate or module `ptr`
##[error] --> library/core/src/ptr/mut_ptr.rs:1924:24
  |
6 | let slice: *mut [i8] = ptr::slice_from_raw_parts_mut(ptr::null_mut(), 3);
  |                        ^^^ use of undeclared crate or module `ptr`
help: a builtin type with a similar name exists
  |
  |
6 | let slice: *mut [i8] = str::slice_from_raw_parts_mut(ptr::null_mut(), 3);
help: consider importing one of these items
  |
5 + use core::ptr;
  |
  |
5 + use std::ptr;
  |

error[E0433]: failed to resolve: use of undeclared crate or module `ptr`
##[error] --> library/core/src/ptr/mut_ptr.rs:1924:54
  |
6 | let slice: *mut [i8] = ptr::slice_from_raw_parts_mut(ptr::null_mut(), 3);
  |                                                      ^^^ use of undeclared crate or module `ptr`
help: a builtin type with a similar name exists
  |
  |
6 | let slice: *mut [i8] = ptr::slice_from_raw_parts_mut(str::null_mut(), 3);
help: consider importing one of these items
  |
5 + use core::ptr;
  |
---
    library/core/src/ptr/mut_ptr.rs - ptr::mut_ptr::*mut[T]::is_empty (line 1921)

test result: FAILED. 4186 passed; 2 failed; 42 ignored; 0 measured; 0 filtered out; finished in 131.11s

error: doctest failed, to rerun pass `-p core --doc`
  local time: Fri Nov 24 07:17:14 UTC 2023
  network time: Fri, 24 Nov 2023 07:17:14 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/ptr/const_ptr.rs
+++ b/library/core/src/ptr/const_ptr.rs
@@ -1644,6 +1644,24 @@ impl<T> *const [T] {
         metadata(self)
     }
 
+    /// Returns `true` if the raw slice has a length of 0.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(slice_ptr_len)]
+    /// use std::ptr;
+    ///
+    /// let slice: *const [i8] = ptr::slice_from_raw_parts(ptr::null(), 3);
+    /// assert!(!slice.is_empty());
+    /// ```
+    #[inline(always)]
+    #[unstable(feature = "slice_ptr_len", issue = "71146")]
+    #[rustc_const_unstable(feature = "const_slice_ptr_len", issue = "71146")]
+    pub const fn is_empty(self) -> bool {
+        self.len() == 0
+    }
+
     /// Returns a raw pointer to the slice's buffer.
     ///
     /// This is equivalent to casting `self` to `*const T`, but more type-safe.

--- a/library/core/src/ptr/mut_ptr.rs
+++ b/library/core/src/ptr/mut_ptr.rs
@@ -1920,10 +1920,10 @@ impl<T> *mut [T] {
     ///
     /// ```
     /// #![feature(slice_ptr_len)]
+    /// use std::ptr;
     ///
-    /// let mut a = [1, 2, 3];
-    /// let ptr = &mut a as *mut [_];
-    /// assert!(!ptr.is_empty());
+    /// let slice: *mut [i8] = ptr::slice_from_raw_parts_mut(ptr::null_mut(), 3);
+    /// assert!(!slice.is_empty());
     /// ```
     #[inline(always)]
     #[unstable(feature = "slice_ptr_len", issue = "71146")]

